### ISSUE TITLE
Questions about the training objectives

### ISSUE THREAD

**Issue by logicwong** [NExT-GPT/NExT-GPT#7]
Hi, thank you for the fantastic work! I have a few questions about this work:

- Have you compared the effectiveness of generating signal tokens versus directly generating captions? In Next-GPT, the representation of "<IMG_0> <IMG_1> <IMG_2> <IMG_3>" is aligned with the outputs of the CLIP text encoder. What if we directly generate the corresponding caption (<IMG_0> {caption} <IMG_1>) and input it to the CLIP text encoder? The motivation behind this is that it is often challenging to optimize the model using a combination of l2 loss and cross-entropy, as controlling the proportion of the two can be difficult. By directly generating captions, we would only need to optimize the entire model through cross-entropy, and eliminate the need for introducing the Output Projection.

- Does introducing L2 loss contribute to improved performance in image-to-text tasks (such as coco caption)? Have you conducted any related ablation experiments?

- I noticed that Next-GPT attained an impressive CIDEr score of 156.7 on COCO, surpassing previous approaches utilizing finetuning and SCST. Did Next-GPT undergo finetuning or SCST on COCO?

**Comment by scofield7419** [comment]
Hi @logicwong, thx for your attention, and wise questions.

- So if directly generating captions as instructions for the downstream decoders, there is no difference between the prior pipeline systems, e.g., Visual-ChatGPT, HuggingGPT. In that way, some simple instructions can be smoothly passed to the decoders (I guess Visual-ChatGPT, HuggingGPT can do well in these), but may fail to deliver some intricate ones, i.e., discrete captioning tokens carry much more limited information than the continuous representations (of signal tokens).
- For the optimization with l2 loss and cross-entropy, you are right, it's challenging to coordinate both. And this has a clear impact on the training performance. But it is still worth trying, careful tuning. From our observation, L2 loss of feature alignment is quite important to the performance of overall.
- The performance of image captioning is quite dependent on the upper-bound of 'encoder-LLM' (ImageBind+Vicuna), and the score indicates this combination is quite effective (thanks to ImageBind). Yes in-domain finetuning is taken, in line with baselines.

**Comment by logicwong** [comment]
Thank you for the detailed reply!
